src/app/backend/handler/apihandler.go

```diff
@@ -271,6 +271,15 @@ func CreateHTTPAPIHandler(iManager integration.IntegrationManager, cManager clie
 		apiV1Ws.GET("/deployment/{namespace}/{deployment}/newreplicaset").
 			To(apiHandler.handleGetDeploymentNewReplicaSet).
 			Writes(replicaset.ReplicaSet{}))
+	apiV1Ws.Route(
+		apiV1Ws.POST("/{kind}/{namespace}/{deployment}/rollback").
+			To(apiHandler.handleDeployRollback).
+			Reads(deployment.RolloutSpec{}).
+			Writes(deployment.RolloutSpec{}))
+	apiV1Ws.Route(
+		apiV1Ws.GET("/{kind}/{namespace}/{deployment}/restart").
```

Comment:
Executing action on a resource should not use GET. It should be POST.

Reply:
done. but change to PUT, following other actions.

Reply:
PUT makes sense as we will somehow mutate the existing deployment.